src/MemoryGraph/MemoryGraph.cs

```diff
@@ -10,7 +10,17 @@ public class MemoryGraph : Graph, IFastSerializable
         public MemoryGraph(int expectedSize)
             : base(expectedSize)
         {
-            m_addressToNodeIndex = new Dictionary<Address, NodeIndex>(expectedSize);
+            // If we have too many addresses we will reach the Dictionary's internal array's size limit and throw.
+            // Therefore use a new implementation of it that is similar in performance but that can handle the extra load.
+            if (expectedSize > 200_000)
```

Comment:
Can you share where this number comes from?  I realize that we have to pick something if we want to only impact "large" dumps, but just want to understand the math behind this.